### DISCUSSION TITLE
Unit tests

### DISCUSSION THREAD

**Discussion by shred86** [humbertogontijo/homeassistant-roborock#81] [General]
@humbertogontijo Not sure if you've already started writing tests but I'd be more than happy to try to start working on them. I'd just focus on the HA integration part for now since I'm assuming the API will end up being its own library at some point. My plan was to use the [pytest-homeassistant-custom-component](https://github.com/MatthewFlamm/pytest-homeassistant-custom-component).

If you’ve already started writing them or are planning to, no worries! You’ll definitely be able to write them much faster than me. 😄

**Comment by humbertogontijo** [comment]
This integration should've had tests from the beginning. Sorry about my laziness, it would be super appreciated. And you are right, the api folder will soon be a library.

**Reply by shred86** [reply]
Ha, no worries! I created a [PR](https://github.com/humbertogontijo/homeassistant-roborock/pull/82) with the start of some tests and minor tweaks). I only submitted the tests I did for the config flow. I spent some time trying to get tests setup for the entities but I was wondering if you were planning on refactoring any of the API. Specifically looking at how `__init__.py` is setting up the integration starting on [line 27,](https://github.com/humbertogontijo/homeassistant-roborock/blob/353cb4a661da88c47de6099a64042b8117ec2579/custom_components/roborock/__init__.py#L27), it seems like there's some code that could just be internal to the API.
